internal/command/views/test.go

```diff
@@ -460,21 +465,24 @@ func (t *TestJSON) DestroySummary(diags tfdiags.Diagnostics, run *moduletest.Run
 
 		if run != nil {
 			t.view.log.Error(
-				fmt.Sprintf("OpenTofu left some resources in state after executing %s/%s, they need to be cleaned up manually.", file.Name, run.Name),
+				fmt.Sprintf("OpenTofu left some resources in state after executing %s/%s, these left-over resources can be viewed by reading the statefile written to disk(errored_test.tfstate).", file.Name, run.Name),
 				"type", json.MessageTestCleanup,
 				json.MessageTestCleanup, cleanup,
 				"@testfile", file.Name,
 				"@testrun", run.Name)
 		} else {
 			t.view.log.Error(
-				fmt.Sprintf("OpenTofu left some resources in state after executing %s, they need to be cleaned up manually.", file.Name),
+				fmt.Sprintf("OpenTofu left some resources in state after executing %s, these left-over resources can be viewed by reading the statefile written to disk(errored_test.tfstate).", file.Name),
 				"type", json.MessageTestCleanup,
 				json.MessageTestCleanup, cleanup,
 				"@testfile", file.Name)
 		}
-
+		t.view.log.Info("Writing state to file: errored_test.tfstate")
 	}
-
+	op := OperationJSON{
+		view: t.view,
+	}
+	diags = diags.Append(SaveErroredTestStateFile(state, &op))
```

Comment:
You are always writing to the errored-state file (even if nothing errored), but you only write the log line `"Writing state to file"` when it errored

Please make sure the state file is only written when there's an error in destroy

Also, maybe adapt the tests to check the existence of the state file

Reply:
This was an oversight on my end, fixed it 👍🏼.  

As for the tests, I am not clear on how you'd like the tests to be changed. Could you please elaborate on this?

Reply:
Right now the tests only check the logs, checking whether or not the logs 
write `"Writing state to file"`, but they are not actually testing whether the file was created or not (and whether or not the state in the file is correct).

Also, right now, the test is testing out the `SaveErroredTestStateFile` function, and not the actual test command, like the other tests in the file. I personally think that `SaveErroredTestStateFile` didn't really have to be a public function.

Ideally, the tests should actually run the test command, and later check whether or not the state file exists, and what it contains. Doing this would have made the tests more precise, and would have detected this issue early on.

However, I would say that doing this is optional, in my opinion, as setting up such a test would not be too simple (would require mocking failure when destroying), so feel free to resolve this comment

Reply:
I understand how testing doing that would make the tests more precise and inclusive overall, but in my opinion, I feel like this would mean unnecessary additional complexities (like mocking failure when destroying). 

Essentially we would be testing out the same function, but with the workflow of "tofu test" running from the start, instead of how `SaveErroredTestStateFile` is currently being tested. I took inspiration from how tests for DestroySummary has been written, I believe that the current testing should suffice.